api/v1beta1/ipaddresspool_types.go

```diff
@@ -39,6 +39,23 @@ type IPAddressPoolSpec struct {
 	// +optional
 	// +kubebuilder:default:=false
 	AvoidBuggyIPs bool `json:"avoidBuggyIPs,omitempty"`
+
+	// AllocateTo makes ip pool allocation to specific namespace and/or service.
+	// +optional
+	AllocateTo ServiceAllocation `json:"serviceAllocation,omitempty"`
+
+	// BGPPeers attaches ip pool to specific bgp peer(s) where its ip addresses
+	// get advertised.
+	// +optional
+	BGPPeers []string `json:"bgpPeers,omitempty"`
```

Comment:
this is not required, as it's covered by the peers specified in the BGPAdvertisement CRD.